src/gui/mainwindow.cpp

```diff
@@ -1437,7 +1437,7 @@ void MainWindow::loadPreferences(bool configureSession)
 #else
     const bool newSystrayIntegration = pref->systrayIntegration();
     m_ui->actionLock->setVisible(newSystrayIntegration);
-    if (newSystrayIntegration != (m_systrayIcon != 0)) {
+    if (newSystrayIntegration != (m_systrayIcon != nullptr)) {
```

Comment:
equal to `newSystrayIntegration == m_systrayIcon`?

Reply:
error: invalid operands to binary expression ('int' and 'QPointer<QSystemTrayIcon>')

Reply:
I give up, I just can't turn my head around and find the correct statement :\
There must be something more direct to express this.

Reply:
The correct is:
```c++
((newSystrayIntegration && !m_systrayIcon) || (!newSystrayIntegration && m_systrayIcon))
```
Since c++ has no logical xor operator...